packages/react/src/components/DataGrid/utils.tsx

```diff
@@ -22,9 +22,9 @@ export const useHeadlessColumns = <T extends Row>({
 }): ColumnDef<T, any>[] => {
   const { t } = useCunningham();
   const columnHelper = createColumnHelper<T>();
-  let headlessColumns = columns.map((column) => {
+  let headlessColumns = columns.map((column, index) => {
     const opts = {
-      id: column.field ?? "actions",
+      id: column.field ?? `actions_${index}`,
```

Comment:
Crystal clear! Since I haven't had the opportunity to work on this particular component, my question may not be directly relevant. Therefore, please feel free to disregard it if it doesn't apply.

I noticed that when reordering the columns, the column's `id` would be updated. However, I'm curious whether this would also trigger a re-rendering of the column content, even if the content itself hasn't changed. In React, using the index as a key for a list is considered an antipattern. I'm wondering if the same principle applies in this scenario as well.

Reply:
Yes it's not perfect, but we don't have too much data to rely on (https://github.com/openfun/cunningham/blob/main/packages/react/src/components/DataGrid/index.tsx#L26-L32), we could eventually improve a bit by doing that:
```
id: column.field ?? `actions-${column.headerName ? `${column.headerName}-${index}` : index }`,
```
Open to suggestion.

Reply:
We could do something like that as well ([dd06f1a](https://github.com/openfun/cunningham/pull/119/commits/dd06f1af524bd4daf644d999e0c40bcd984154b1)), an `id` property is mandatory if the property `renderCell` is defined. The bad side is that it creates a breaking change with the datagrid with renderCell defined.